mm2src/coins_activation/src/tendermint_with_assets_activation.rs

```diff
@@ -285,7 +285,16 @@ impl PlatformWithTokensActivationOps for TendermintCoin {
         self.spawner().spawn_with_settings(fut, settings);
     }
 
-    fn handle_balance_streaming(&self, config: &EventStreamConfiguration) {
-        EventBehaviour::spawn_if_active(self.clone(), config);
+    async fn handle_balance_streaming(
+        &self,
+        config: &EventStreamConfiguration,
+    ) -> Result<(), MmError<Self::ActivationError>> {
+        if let EventInitStatus::Failed(err) = EventBehaviour::spawn_if_active(self.clone(), config).await {
```

Comment:
`spawn_if_active` implementation for tendermint will never return `EventInitStatus::Failed`. Shouldn't we return `EventInitStatus::Failed` below in case of an error instead of using `expect` https://github.com/KomodoPlatform/komodo-defi-framework/blob/dbb2b22cc28dafda36052c727c1837e2b0a76f57/mm2src/coins/tendermint/tendermint_balance_events.rs#L173

Reply:
We do return: https://github.com/KomodoPlatform/komodo-defi-framework/blob/d7bd0d701d046e5061b176514ad0d27aaf3a30d9/mm2src/coins/tendermint/tendermint_balance_events.rs#L39-L41

Reply:
`.expect("Event initialization status must be recieved.")` has different purpose, it's for the channels result which only fails sender/reciever drops

Reply:
I did this change https://github.com/KomodoPlatform/komodo-defi-framework/pull/1978#discussion_r1350342005 locally before reviewing this part so I didn't find it 😅 
I will leave this until the comment here https://github.com/KomodoPlatform/komodo-defi-framework/pull/1978#discussion_r1350352862 is resolved

Reply:
P.S. I still think we should return an error or `EventInitStatus::Failed` instead of `expect` below
https://github.com/KomodoPlatform/komodo-defi-framework/blob/dbb2b22cc28dafda36052c727c1837e2b0a76f57/mm2src/coins/tendermint/tendermint_balance_events.rs#L173